### ISSUE TITLE
Implement segmentation algorithm to replace CBS.

### ISSUE THREAD

**Issue by samuelklee** [broadinstitute/gatk-protected#334]
For example, the Bayesian change point (bcp) R package.

**Comment by samuelklee** [comment]
Notes on a few possibilities I've explored:

bcp (Bayesian change points): 
-pros: R package available, probabilistic breakpoints, fast (linear), multivariate
-cons: assumes normal distribution, a few free parameters (but standard values typically taken)
ecp:
-pros: R package available, non-parametric, multivariate (even seems to segment well on [alt, ref, tn cr])
-cons: very slow (quadratic, ~20 min/chr over het-only coverage)
tvd (total variation denoising):
-pros: R package available (java implementation should also be straightforward), fast, single penalty parameter
-cons: not clear how to do our particular multivariate case, smoothing step only (need segmentation step after)
PELT (pruned exact linear time):
-pros: R package available (java implementation should also be straightforward), fast (linear), single penalty parameter, can use our CR/AF likelihoods
-cons: some papers seem to suggest that oversegmentation is an issue
wbs (wild binary segmentation):
-pros: R package available, fast, some advantages over CBS
-cons: some underlying Gaussian assumptions, free parameters/choice of stopping criteria

I've explored a few additional possibilities, including PDP, NP-PELT, and a few other Bayesian approaches.  Although something multivariate + Bayesian would be great, in practice I think these would be difficult to implement (for example, it's not clear how to handle the "missing data" problem, i.e., the fact that targets and hets do not overlap 1-to-1 to give a nice multivariate quantity to segment on).

My feeling is that something simple like univariate PELT + our CR/AF likelihoods + segment union could yield good results, while being fast, easy to implement in java, and relatively parameter free (requiring only a single parameter that penalizes changepoints---we'd still need to choose this via a parameter sweep/cross validation, which would require a truth set).  

Ultimately, if we implement an HMM hooked up to our likelihoods that gives good results and can account for missing data, then we can get rid of this whole segment + segment-union paradigm.  Again a segmentation truth set will be critical here.
